### PR TITLE
[WIP] language selection on config screen

### PR DESCRIPTION
**Pull request by nicosomb**
Need one test on listener. I wait for @j0k3r and the other PR.

### CODE REVIEW

File: src/Wallabag/CoreBundle/Controller/ConfigController.php [hunk 1]
```diff
@@ -41,7 +41,7 @@ public function indexAction(Request $request)
 
             $this->get('session')->getFlashBag()->add(
                 'notice',
-                'Config saved'
+                'Config saved. Some parameters will be considered after disconnection.'
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Controller/ConfigController.php]: Don't forget to update the translation after that.

File: src/Wallabag/CoreBundle/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+
+class Configuration implements ConfigurationInterface
+{
+    public function getConfigTreeBuilder()
+    {
+        $treeBuilder = new TreeBuilder();
+        $rootNode = $treeBuilder->root('wallabag_core');
+
+        $rootNode->
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/DependencyInjection/Configuration.php]: I prefer if we keep `->` before the _action_.
So:

```php
$rootNode
    ->children()

File: src/Wallabag/CoreBundle/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+            children()
+                ->arrayNode('languages')
+                ->prototype('scalar')->end()
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/DependencyInjection/Configuration.php]: You missed one indentation here

File: src/Wallabag/CoreBundle/DependencyInjection/WallabagCoreExtension.php [hunk 4]
```diff
@@ -11,6 +12,11 @@ class WallabagCoreExtension extends Extension
 {
     public function load(array $configs, ContainerBuilder $container)
     {
+        $processor = new Processor();
+        $configuration = new Configuration();
+        $config = $processor->processConfiguration($configuration, $configs);
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/DependencyInjection/WallabagCoreExtension.php]: You can avoid `$processor` and use `$this->processConfiguration(` directly.

File: src/Wallabag/CoreBundle/Form/Type/ConfigType.php [hunk 5]
```diff
@@ -9,16 +9,20 @@
 class ConfigType extends AbstractType
 {
     private $themes = array();
+    private $languages = array();
 
     /**
-     * @param array $themes Themes come from the LiipThemeBundle (liip_theme.themes)
+     * @param array $themes    Themes come from the LiipThemeBundle (liip_theme.themes)
+     * @param array $languages Languages come from configuration
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Form/Type/ConfigType.php]: Could you add more information about what should be `$languages` (an array with code as key and label as value)

File: src/Wallabag/CoreBundle/Controller/ConfigController.php (hunk 1 above)
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Controller/ConfigController.php]: Never mind :)

File: src/Wallabag/CoreBundle/Tests/Controller/ConfigControllerTest.php [hunk 6]
```diff
@@ -65,12 +65,7 @@ public function dataForUpdateFailed()
             array(array(
                 'config[theme]' => 0,
                 'config[items_per_page]' => '',
-                'config[language]' => 'fr_FR',
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Tests/Controller/ConfigControllerTest.php]: Why did you remove the `fr` test case?

Comment (nicosomb) [inline src/Wallabag/CoreBundle/Tests/Controller/ConfigControllerTest.php]: I changed `fr` to `en` to not change all test where we looked for english text. 
I removed the test where language is empty, because it failed. And as you didn't write a test where `config[theme]` is empty, I removed this one. Don't know if I'm clear.

**Comment by nicosomb** [comment]
any idea on how can I write a test for this feature?

**Comment by j0k3r** [comment]
Did you see that test https://github.com/wallabag/wallabag/blob/v2/src/Wallabag/CoreBundle/Tests/EventListener/RegistrationConfirmedListenerTest.php ?

**Comment by nicosomb** [comment]
I will have a look.

**Comment by nicosomb** [comment]
Thank you @j0k3r for the test. If it's good for you, we can merge.
